### DISCUSSION TITLE
Can't load new firmware

### DISCUSSION THREAD

**Discussion by reinhardtjh** [thehighnibble/thehighnibble.github.io#76] [VT132 terminal emulation]
I have a brand new build of a kit I bought 2 years ago.  It's been a bit of a learning curve but I got into the setup screens and have figured out the setting.  My firmware is old (V1.2.0) so I want to update it.

I've tried the OTA firmware upload but it's not working.  I can connect to my wifi but I get the following

AT+U?
OTA BEGIN FAILED

If I try the update anyway I get:
AT+U^
OTA NO IMAGE

The Wi-Fi connected:
AT+W?
WIFI CONNECTED

The interface has gotten an IP:
AT+W$
172.16.255.220

AT+U$
RUNNING  [app0]  [VT132 V1.2.0] [00010000:0014ffff]
BOOT        [app0]  [VT132 V1.2.0] [00010000:0014ffff]

Any ideas where to go next?  It there a problem on the GitHub side?

**Comment by reinhardtjh** [comment]
And I found the answer.  It's the certificates, I assume.  I found the old Google Group list and sure enough, this same problem happened a year ago in 2022.

https://groups.google.com/g/imsai8080esp/c/AYOzgUc2ZYg/m/dil_YBj3BQAJ

I've verified the shortened URL doesn't work.  I didn't expect it to, but I hoped.
Looks like I have to figure out another way of updating.

**Comment by dmcnaugh** [comment]
Yes, the Amazon S3 and GitHub certificates where the reason.
This is a known issue and was fixed in v1.3.0 of the firmware.
The release notes that accompany that version provide details of how to work around the issue.
See: https://github.com/thehighnibble/vt132/releases/tag/V1.3.0
All release notes since then have referred you to this solution in the opening paragraph.

**Reply by reinhardtjh** [reply]
Except that using the shortened URL didn't work.

AT+W+ 
WIFI RECONNECTED

OK
AT+U=https://bit.ly/VT132latest
OTA URL SET

OK
AT+U?
OTA BEGIN FAILED

Did I miss something?

**Reply by dmcnaugh** [reply]
I'm not sure. It was a while ago that I set this up to work.
I'd have to flash a VT132 with v1.2.0 and try this again myself to diagnose.
I'll be able to try that on the weekend.

Meanwhile I'd recommend using one of the other methods: maybe a local web server - usually not to hard to set up.

**Reply by reinhardtjh** [reply]
Thanks for your help!  I've got a TTL Serial cable around here that I will be using to try to flash it next.

I do know the short URL works.  If I put it in my browser it tries to download the VT132.BIN file.

**Reply by reinhardtjh** [reply]
I set up the Apache web server on my Mac and was able to load V1.5.0 with the local web server option.  

Thanks!

**Reply by dmcnaugh** [reply]
I'm glad you got it working.
